craft_providers/lxd/remotes.py

```diff
@@ -15,50 +16,165 @@
+
+from .errors import LXDError
 from .lxc import LXC
 
 logger = logging.getLogger(__name__)
 
-BUILDD_REMOTE_NAME = "craft-com.ubuntu.cloud-buildd"
-BUILDD_REMOTE_ADDR = "https://cloud-images.ubuntu.com/buildd/releases"
+BUILDD_RELEASES_REMOTE_NAME = "craft-com.ubuntu.cloud-buildd"
+BUILDD_RELEASES_REMOTE_ADDRESS = "https://cloud-images.ubuntu.com/buildd/releases"
 
+# XXX: lunar and kinetic buildd daily images are not working (LP #2007419)
+BUILDD_DAILY_REMOTE_NAME = "craft-com.ubuntu.cloud-buildd-daily"
+BUILDD_DAILY_REMOTE_ADDRESS = "https://cloud-images.ubuntu.com/buildd/daily"
 
-def configure_buildd_image_remote(
-    lxc: LXC = LXC(),
-) -> str:
-    """Configure buildd remote, adding remote as required.
+# temporarily use the cloud release images until daily buildd images are fixed
+DAILY_REMOTE_NAME = "ubuntu-daily"
+DAILY_REMOTE_ADDRESS = "https://cloud-images.ubuntu.com/daily"
 
-    :param lxc: LXC client.
 
-    :returns: Name of remote to pass to launcher.
+class ProtocolType(Enum):
+    """Enumeration of protocols for LXD remotes."""
+
+    LXD = "lxd"
+    SIMPLESTREAMS = "simplestreams"
+
+
+@dataclass
+class RemoteImage:
+    """Contains the name, location, and details of a remote LXD image.
+
+    :param image_name: Name of the image on the remote (e.g. `core22` or `lunar`).
+    :param remote_name: Name of the remote server.
+    :param remote_address: Address of the remote (can be an IP, FDQN, URL, or token)
+    :param remote_protocol: Remote protocol (options are `lxd` and `simplestreams`)
+    :param is_stable: True if the image is a stable release. Daily and devel images
+    are not stable.
     """
-    if BUILDD_REMOTE_NAME in lxc.remote_list():
-        logger.debug("Remote %r already exists.", BUILDD_REMOTE_NAME)
-    else:
-        try:
-            lxc.remote_add(
-                remote=BUILDD_REMOTE_NAME,
-                addr=BUILDD_REMOTE_ADDR,
-                protocol="simplestreams",
-            )
-        except Exception as exc:  # pylint: disable=broad-except
-            # the remote adding failed, no matter really how: if it was because a race
-            # condition on remote creation (it's not idempotent) and now the remote is
-            # there, the purpose of this function is done (otherwise we let the
-            # original exception fly)
-            if BUILDD_REMOTE_NAME in lxc.remote_list():
-                logger.debug(
-                    "Remote %r is present on second check, ignoring exception %r.",
-                    BUILDD_REMOTE_NAME,
-                    exc,
+
+    image_name: str
+    remote_name: str
+    remote_address: str
+    remote_protocol: ProtocolType
+    is_stable: bool
+
+    def add_remote(self, lxc: LXC = LXC()) -> None:
+        """Add the LXD remote for an image.
+
+        If the remote already exists, it will not be re-added.
+
+        :param lxc: LXC client.
+        """
+        if self.remote_name in lxc.remote_list():
```

Comment:
Theoretically this means the user could modify the remote and cause problems (images not available, wrong images coming, etc.). It might be worth checking the remote address in that case and providing a warning if it differs. (Or maybe that's too much of an edge case to care about considering that this is the same logic as previously existed.)

Reply:
You're right - it's a good check and an edge case.  I'll add a TODO for now